Add info that Vocab and StringStore are not static in docs (#12427)
explosion/spaCy: website/docs/api/stringstore.mdx

integer IDs. This ensures that strings always map to the same ID, even from
different `StringStores`.

<Infobox variant="warning">

> Please note that the `StringStore` size is not static and increases as texts are
> processed and new tokens are seen.

Reviewer: I feel like this is clearer, but feel free to ignore & resolve if you disagree.
```suggestion
Please note that `StringStore` instances are not static. It increases in size as 
texts with new tokens are processed.
```
